server/test/unit/server/async/test_scheduler.py

```diff
@@ -282,6 +284,17 @@ def test_calls_handle_heartbeat(self, mock_heartbeat, mock_tick):
 
         mock_heartbeat.assert_called_once()
 
+    @mock.patch('celery.beat.Scheduler.__init__', new=mock.Mock())
+    @mock.patch('celery.beat.Scheduler.tick')
+    def test_heartbeat_lock(self, mock_tick):
```

Comment:
There needs to be a test for each codepath you have added. There are at least 4 because there is an if/else and try/except so used together that is 4 codepaths. I recommend putting them all on the TestSchedulerTick class.